lib/dry/configurable/dsl.rb

```diff
@@ -77,8 +141,17 @@ def ensure_valid_options(options)
         return if options.none?
 
         invalid_keys = options.keys - Setting::OPTIONS
+
         raise ArgumentError, "Invalid options: #{invalid_keys.inspect}" unless invalid_keys.empty?
       end
+
+      # Returns a tuple of valid and invalid options hashes derived from the options hash
+      # given to the setting
+      def valid_and_invalid_options(options)
+        options.each_with_object([{}, {}]) do |(key, val), (valid_options, invalid_options)|
+          (Setting::OPTIONS.include?(key) ? valid_options : invalid_options)[key] = val
+        end
```

Comment:
```ruby
options.partition { |k, _| Setting::OPTIONS.include?(k) }.map(&:to_h)
```

Reply:
So much neater, thanks @flash-gordon! I've applied this now.